Big Update to V3

Mettwurstler opened the pull request:
Hello :) 

I tried to create a new updated version as a standalone app based on the ideas from notablackbear. I never coded with python before, so it was a nice opportunity to learn python a little bit :D I actually use my program ingame for little chaos on top and it works pretty well. I hope it will get the job done when you use it too.

Mettwurstler committed:
Delete CeruleanRing.png

Mettwurstler committed:
Delete CurrencyRerollRare.png

Mettwurstler committed:
Delete Gui_Button_V1.ui

Mettwurstler committed:
Delete Gui_Button_V2.ui

Mettwurstler committed:
Delete POEQOL_Base.filter

Mettwurstler committed:
Delete POE_QOL2.bak.exe

Mettwurstler committed:
Delete POE_QOL2.exe

Mettwurstler committed:
Delete POE_QOL2.py

Mettwurstler committed:
Delete POE_QOL_v2-0-3.exe

Mettwurstler committed:
Delete README.md

Mettwurstler committed:
Delete Setup.ini

Mettwurstler committed:
Delete VermillionRing.png

Mettwurstler committed:
Delete blighted map.png

Mettwurstler committed:
Delete chaos_items_filter.filter

Mettwurstler committed:
Delete poeqol2_logfile.txt

Mettwurstler committed:
Delete veiled.png

Mettwurstler committed:
Add files via upload

Mettwurstler committed:
Add files via upload

Mettwurstler committed:
Create README.md

Mettwurstler committed:
Update README.md

Mettwurstler committed:
Update README.md

Mettwurstler committed:
Delete POE_QOL_V3.exe

Mettwurstler committed:
Delete PoE_QoL_V3.py

Mettwurstler committed:
Delete PoE_QoL_V3_no_UI.py

Mettwurstler committed:
Add files via upload

Mettwurstler committed:
Update README.md

Mettwurstler committed:
Delete POE_QOL_V3.exe

Mettwurstler committed:
Delete PoE_QoL_V3.py

Mettwurstler committed:
Delete PoE_QoL_V3_no_UI.py

Mettwurstler committed:
Add files via upload

Mettwurstler commented:
UPDATE:
Fixed Problem with the new patch and the new location of the highlights

azra1l commented:
hi there,

you might want to check your code, my avira instantly killed the POE_QOL_V3.exe due to malware detection. so i went to www.virustotal.com and uploaded the file, there is definetly something edgy about this, 16 different virus engines reported some kind of malware:

https://www.virustotal.com/gui/file/ea0813ca72c48406dc7915745ce90540285d752ac054702eb94102158a87c711

you'll have to do some silly captcha or sign up to see the report.

i think that's rather unlikely to be a false-positive. did you include any 3rd-party content in your version?

Mettwurstler commented:
hey :)

my windows defender also kills the .exe when i convert my .py, dont know why :( 
I only wrote the program in PyCharm, included some .imgs, one .gif and one .ico file, thats all. Actually the app reads the stash-tabs with the Path of Exile API. maybe thats the point?

Mettwurstler commented:
I did a quick research and the original tool from Path of Matth is detected as the exact same "virus", see here: https://youtu.be/EYRoGX13nbI?t=40

azra1l commented:
i doubt so, because the v2 does the same and isn't flagged. 
uploaded that file as well to virustotal, it has one random detection from some virus engine i never heard before, i'd consider this a possible false-positive.

i am not too hooked into python programming, maybe check with python reddit folks if they can help you figure out what's going on there...

no offense, but as it is right now, i don't feel exactly safe to use your version.

that said, i also had some plans to make some changes to the app
(basically i want to pimp this and extend it towards the ultimate stash management / vendor recipe helper tool)
might i ask, what did you changed from v2? based on file changes, i'd assume new gui/graphic assets?

Mettwurstler commented:
oh ok, then I will look why it is detected as virus and try to fix this. 

I completely understand you, I dont would use it ether xD

oh nice :) I wrote the gui/graphics almost completely new (nice darkmode). I changed some positions from buttons, so that you can do chaos recipe with the overlay alone. then I implemented the settings in the gui, so you have a settings-window which saves the data and no extra file. additionally i optimized the code, so items will be shown sorted from right to left in stash and I added some additional buttons to search for items in stash like six-links. this works now completely with data from the API, so no copy and paste anymore. 
I think that are the mayor changes. :)

here are some quick screenshots:
![Unbenannt2](https://user-images.githubusercontent.com/72079767/98281639-49001200-1f9d-11eb-9910-5e897ac293a0.PNG)
![Unbenannt](https://user-images.githubusercontent.com/72079767/98281649-4bfb0280-1f9d-11eb-8f13-3a74cf29965e.PNG)

azra1l commented:
that looks smooth. good luck with your code 👍

Mettwurstler commented:
ty 👍

Mettwurstler commented:
puuuuh,
I searched and tried several solutions from google, reddit to fix the problem with the virus detection but nothing works. when I dont bundle the .py file I "only" reach 11 detections from virus engines.. Maybe I will search later some more or I just use it for myself. So annoying... 👎

notablackbear commented:
@Mettwurstler thanks I am picking this up again since league is pretty dead. If you want to contribute here, that's great. I'll look at some of your changes. We have another branch we started on that was a huge backend overhaul. I like what I see here on the front end. Think we can combine the two? I'll take some closer looks.

Mettwurstler commented:
@notablackbear hey, ty :) sure, so maybe you can look at my version more detailed and then we can discuss what would be the best :) i think its optimal to have only one big up-to-date version, so merging the versions sounds nice. in the meantime i detected and patched some minor problems with filtering the correct items from the api but dont uploaded it yet. but this shouldnt change the major code :)

notablackbear commented:
@Mettwurstler I love it! It is working for me.

I love that you did the setting dialogue, and I _actually like the overlay theme_, and I am usually a stickler for that, since I tend to prefer color-blind friendly colors, but I think this is okay.

I am making notes on what is not working and will post them here.

What I would like to do, is make incremental changes and break up the file if possible. I'd like to see about taking this GUI into a separate file and have it imported into the main program when necessary. I think that way we can keep changes in the GUI from breaking things in the back end a little more reliably. Do you think we can do that? This should make it easier to combine @DavidEl03 's backend (#58) with your new interface.

Also, since you said you were kind of new to python, I am not sure if you are also new to Git (it seems like you know GitHub pretty well, so probably not). I think I'd prefer changing one thing at a time, since a whole new file is a lot to swallow. Since the code is pretty bad to start, I think we get a good foundation now and then move forward a little bit at a time.

When you get a chance feel free to push the minor changes you made and I'll pull it to see if it fixes some of the issues I am having

message me on discord to chat live. notaspy #6561

Mettwurstler committed:
Delete PoE_QoL_V3.py

Mettwurstler committed:
Delete POE_QOL_V3.exe

Mettwurstler committed:
Delete PoE_QoL_V3_no_UI.py

Mettwurstler committed:
Add files via upload

Mettwurstler commented:
hey, sorry I had a job interview yesterday and my mind was elsewhere :D sure, breaking the code in some parts is generally a good idea. Its a big pill meanwhile, specially for all who dont wrote the code. 

Yes im completely new to git(-hub) too. I just kept hoping that I did everything right xD 
I just updated my actual state to my branch. I also updated my release 3.1.

Numer1sch commented:
Hey Mettwurstler, thank you for your work. 
I have tested the programm yesterday and have a issue with it. I used the newest regular filter from filterblade. The programm inserst the chaos filter lines correctly but for any reason they get overwritten by other rules in the filter. I have attached the used filter. Maybe you can look into it? Thank you very mutch!
[Regular Filter.txt](https://github.com/notablackbear/poe_qolV2/files/5775539/Regular.Filter.txt)
EDIT: 
Same problem with the semi strict filter from filterblade.

Mettwurstler commented:
hey numer1sch :)

I personally use my own item filter files, so it is possible that it dont functions well with filterblade without changing anything.
just try to copy the chaos filter lines over the section for rare items which start with [3200]. Here all rare items become customised, so a later change won't do anything to them. 
The insertion of the lines at the end is more a save "start place" in the filter. Then you must copy/past it one time depending on your filter until it works correctly.

I hope it works with this little change. gl in the new league and have fun :D

zakaoai commented:
On Filterblade there is an overwrite section don't know if the chaos item filter is better at this place
